src/shogun/labels/LabelEncoder.h

```diff
@@ -0,0 +1,106 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Yuhui Liu
+ *
+ */
+
+#ifndef _LABELENCODER__H__
+#define _LABELENCODER__H__
+
+#include <algorithm>
+#include <map>
+#include <memory>
+#include <set>
+#include <shogun/base/SGObject.h>
+#include <shogun/lib/SGVector.h>
+namespace shogun
+{
+
+	class LabelEncoder : public SGObject
```

Comment:
Should this inherit from `Transformer`?

Reply:
It seems like `Transformer` transform `Features`, not `Labels`.
https://github.com/shogun-toolbox/shogun/blob/41b6247f0a8c0f9ce45b388ae085d42d90bdd16d/src/shogun/transformer/Transformer.h#L60-L62

Reply:
Ah makes sense, nvm!

Reply:
"Implements a reversible mapping from any form of labels to one of Shogun's target label spaces (binary, multi-class, etc)."